### PR TITLE
Gauge chart

### PR DESCRIPTION
**Pull request by lucamattiazzi**
### Updates
adds a gauge as new chart, with the possibility to set the arc size
do not merge, since it would be merged on top of tabular-data-format, just to check if the code looks fine to you since it's my first pr on this project

### Demo screenshot or recording

image
image
image

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Meets the [code style guide](https://github.com/IBM/carbon-charts/wiki/Code-style-guide)
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: packages/core/demo/data/index.ts [hunk 1]
```diff
@@ -61,6 +63,11 @@ export const chartTypes = {
 		angular: "ibm-pie-chart",
 		vue: "ccv-pie-chart"
 	},
+	GaugeChart: {
+		vanilla: "GaugeChart",
+		angular: "ibm-donut-chart",
+		vue: "ccv-donut-chart"
```
Comment (lucafalasco) [inline packages/core/demo/data/index.ts]:
```suggestion
		angular: "ibm-gauge-chart",
		vue: "ccv-gauge-chart"
```

File: packages/core/src/components/graphs/gauge.ts [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+	}
+
+
+	getInnerRadius() {
+		// Compute the outer radius needed
+		const radius = this.computeRadius();
+		return radius * (3 / 4);
+	}
+
+	centerNumberTween(d3Ref, value: number) {
```
Comment (lucafalasco) [inline packages/core/src/components/graphs/gauge.ts:194]: looks a bit weird to me passing `d3Ref` here, maybe rearrange this func in order to pass and return just the value?

File: packages/core/stories/all.stories.ts [hunk 3]
```diff
@@ -144,10 +144,9 @@ if (process.env.NODE_ENV !== "production") {
 		storyUtils.addRadioButtonEventListeners(container);
 
 		const getNewRow = () => {
-			const row = document.createElement("div");
-			row.className = "bx--row";
-
-			return row;
```
Comment (lucafalasco) [inline packages/core/stories/all.stories.ts:150]: wasn't working?

Comment (lucamattiazzi) [reply packages/core/stories/all.stories.ts:150]: it's just to be more coherent with previous code in the same file a few lines above

File: packages/core/src/components/graphs/gauge.ts (hunk 2 above)
Comment (lucamattiazzi) [reply packages/core/src/components/graphs/gauge.ts:194]: it looked weird to me too, but I simply kept the same function from the donut chart, if there is no problem in changing it I'll do

File: packages/core/demo/data/index.ts (hunk 1 above)
Comment (lucamattiazzi) [reply packages/core/demo/data/index.ts]: yep forgot about those one

**Comment by caesarsol** [comment]
What? I never closed the pr...

**Comment by caesarsol** [comment]
ok I got it, I actually deleted the to-merge branch.
anyhow we can check the PR, then make it on the IBM repo :)

**Pull request closed without merging**